src/cli/restore/onedrive_test.go

```diff
@@ -39,73 +37,56 @@ func (suite *OneDriveUnitSuite) TestAddOneDriveCommands() {
 	for _, test := range table {
 		suite.Run(test.name, func() {
 			t := suite.T()
-
-			cmd := &cobra.Command{Use: test.use}
-
-			// persistent flags not added by addCommands
-			flags.AddRunModeFlag(cmd, true)
-
-			c := addOneDriveCommands(cmd)
-			require.NotNil(t, c)
-
-			// non-persistent flags not added by addCommands
-			flags.AddAllProviderFlags(c)
-			flags.AddAllStorageFlags(c)
-
-			cmds := cmd.Commands()
-			require.Len(t, cmds, 1)
-
-			child := cmds[0]
-			assert.Equal(t, test.expectUse, child.Use)
-			assert.Equal(t, test.expectShort, child.Short)
-			tester.AreSameFunc(t, test.expectRunE, child.RunE)
-
-			flagsTD.WithFlags(
-				cmd,
-				oneDriveServiceCommand,
-				[]string{
-					"--" + flags.RunModeFN, flags.RunModeFlagTest,
-					"--" + flags.BackupFN, flagsTD.BackupInput,
-					"--" + flags.FileFN, flagsTD.FlgInputs(flagsTD.FileNameInput),
-					"--" + flags.FolderFN, flagsTD.FlgInputs(flagsTD.FolderPathInput),
-					"--" + flags.FileCreatedAfterFN, flagsTD.FileCreatedAfterInput,
-					"--" + flags.FileCreatedBeforeFN, flagsTD.FileCreatedBeforeInput,
-					"--" + flags.FileModifiedAfterFN, flagsTD.FileModifiedAfterInput,
-					"--" + flags.FileModifiedBeforeFN, flagsTD.FileModifiedBeforeInput,
-
-					"--" + flags.CollisionsFN, flagsTD.Collisions,
-					"--" + flags.DestinationFN, flagsTD.Destination,
-					"--" + flags.ToResourceFN, flagsTD.ToResource,
-
-					// bool flags
-					"--" + flags.NoPermissionsFN,
+			parent := &cobra.Command{Use: restoreCommand}
+
+			cmd := cliTD.SetUpCmdHasFlags(
+				t,
+				parent,
+				addOneDriveCommands,
+				[]cliTD.UseCobraCommandFn{
+					flags.AddAllProviderFlags,
+					flags.AddAllStorageFlags,
 				},
-				flagsTD.PreparedProviderFlags(),
-				flagsTD.PreparedStorageFlags())
-
-			cmd.SetOut(new(bytes.Buffer)) // drop output
-			cmd.SetErr(new(bytes.Buffer)) // drop output
-
-			err := cmd.Execute()
-			assert.NoError(t, err, clues.ToCore(err))
+				flagsTD.WithFlags(
+					oneDriveServiceCommand,
+					[]string{
+						"--" + flags.RunModeFN, flags.RunModeFlagTest,
+						"--" + flags.BackupFN, flagsTD.BackupInput,
+						"--" + flags.FileFN, flagsTD.FlgInputs(flagsTD.FileNameInput),
+						"--" + flags.FolderFN, flagsTD.FlgInputs(flagsTD.FolderPathInput),
+						"--" + flags.FileCreatedAfterFN, flagsTD.FileCreatedAfterInput,
+						"--" + flags.FileCreatedBeforeFN, flagsTD.FileCreatedBeforeInput,
+						"--" + flags.FileModifiedAfterFN, flagsTD.FileModifiedAfterInput,
+						"--" + flags.FileModifiedBeforeFN, flagsTD.FileModifiedBeforeInput,
+						"--" + flags.CollisionsFN, flagsTD.Collisions,
+						"--" + flags.DestinationFN, flagsTD.Destination,
+						"--" + flags.ToResourceFN, flagsTD.ToResource,
+						"--" + flags.NoPermissionsFN,
```

Comment:
These set is common for onedrive, sp and groups. Maybe we can collect it to a global var.